engine-types/src/account_id.rs

```diff
@@ -93,6 +92,20 @@ impl AccountId {
     }
 }
 
+impl BorshDeserialize for AccountId {
+    fn deserialize(buf: &mut &[u8]) -> io::Result<Self> {
+        let account: String = BorshDeserialize::deserialize(buf)?;
+
+        // It's for saving backward compatibility.
+        if account.is_empty() {
+            return Ok(Self::default());
+        }
```

Comment:
Pay attention to this approach, please. It seems should be ok because we just skip validating empty strings only... which is the default one in our case. All non-empty strings will be validated. 
cc @joshuajbouw @birchmd

Reply:
> we are kind of stuck now

Yes. Exactly.

Reply:
We don't need to persist it, we can't reply all transactions without exceptions already. We've fixed all those issues. Imo, safe to remove it because we do plan on going to NEAR's AccountId crate at some point. I do understand that we would need a new snapshot however though as those as all default Account Id's?

Reply:
It's not just about old transactions, I think it is still true that `bridge_prover_id` is an empty string on mainnet. It's not used, so this isn't a problem for anything, but this does meant that some kind of migration would be needed if we wanted to make the empty string an invalid value.

I also think that if we do decide to do that migration we should still maintain backwards compatibility because it would not be acceptable to be unable to replay such a large portion of the Aurora mainnet history. I think this can be achieved by being careful in how we define the legacy version of the engine state when we do the migration.

Reply:
But near's account_id doesn't have the `Default` implementation.